src/script/interpreter.cpp

```diff
@@ -379,7 +379,45 @@ bool EvalScript(vector<vector<unsigned char> >& stack, const CScript& script, un
                     break;
                 }
 
-                case OP_NOP1: case OP_NOP3: case OP_NOP4: case OP_NOP5:
+                case OP_CHECKSEQUENCEVERIFY:
+                {
+                    if (!(flags & SCRIPT_VERIFY_CHECKSEQUENCEVERIFY)) {
+                        // not enabled; treat as a NOP3
+                        if (flags & SCRIPT_VERIFY_DISCOURAGE_UPGRADABLE_NOPS) {
+                            return set_error(serror, SCRIPT_ERR_DISCOURAGE_UPGRADABLE_NOPS);
+                        }
+                        break;
+                    }
+
+                    if (stack.size() < 1)
+                        return set_error(serror, SCRIPT_ERR_INVALID_STACK_OPERATION);
+
+                    // nSequence, like nLockTime, is a 32-bit unsigned integer
+                    // field. See the comment in CHECKLOCKTIMEVERIFY regarding
+                    // 5-byte numeric operands.
+                    const CScriptNum nSequence(stacktop(-1), fRequireMinimal, 5);
+
+                    // In the rare event that the argument may be < 0 due to
+                    // some arithmetic being done first, you can always use
+                    // 0 MAX CHECKSEQUENCEVERIFY.
+                    if (nSequence < 0)
+                        return set_error(serror, SCRIPT_ERR_NEGATIVE_LOCKTIME);
+
+                    // To provide for future soft-fork extensibility, if the
+                    // operand is too large to be treated as a relative lock-
+                    // time, CHECKSEQUENCEVERIFY behaves as a NOP.
+                    if (nSequence >= CTxIn::SEQUENCE_LOCKTIME_THRESHOLD)
+                        break;
+
+                    // Actually compare the specified inverse sequence number
```

Comment:
This email I got! Thanks I make the correction.
On Oct 1, 2015 11:12 AM, "฿tcDrak" <notifications@github.com> wrote:

> In src/script/interpreter.cpp
> <https://github.com/bitcoin/bitcoin/pull/6564#discussion_r40946716>:
>
> > +                    // 5-byte numeric operands.
> > +                    const CScriptNum nSequence(stacktop(-1), fRequireMinimal, 5);
> > +
> > +                    // In the rare event that the argument may be < 0 due to
> > +                    // some arithmetic being done first, you can always use
> > +                    // 0 MAX CHECKSEQUENCEVERIFY.
> > +                    if (nSequence < 0)
> > +                        return set_error(serror, SCRIPT_ERR_NEGATIVE_LOCKTIME);
> > +
> > +                    // To provide for future soft-fork extensibility, if the
> > +                    // operand is too large to be treated as a relative lock-
> > +                    // time, CHECKSEQUENCEVERIFY behaves as a NOP.
> > +                    if (nSequence >= CTxIn::SEQUENCE_LOCKTIME_THRESHOLD)
> > +                        break;
> > +
> > +                    // Actually compare the specified inverse sequence number
>
> s/inverse//
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bitcoin/bitcoin/pull/6564/files#r40946716>.
>